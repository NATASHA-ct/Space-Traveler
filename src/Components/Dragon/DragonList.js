import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchAllDragon } from '../../redux/dragons/dragon';
import Dragon from './Dragon';
import './Dragon.css';

const DragonList = () => {
  const dispatch = useDispatch();
  const dragons = useSelector((state) => state.dragons);

  useEffect(() => {
    if (!dragons.length) dispatch(fetchAllDragon());
  });

  const list = dragons.map((item) => (
    <Dragon
      key={item.id}
      id={item.id}
      name={item.name}
      type={item.type}
      img={item.img}
      reservation={item.reservation}
    />
  ));
  return <>{list}</>;
};
export default DragonList;
