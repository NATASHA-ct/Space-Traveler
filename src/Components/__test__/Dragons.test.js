import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";
import DragonList from "../Dragon/DragonList";
import store from "../../redux/configureStore";
describe("dragon test", () => {
  test("snapshot for dragon", () => {
    const tree = render(
      <React.StrictMode>
        <Provider store={store}>
          <Router>
            <DragonList />
          </Router>
        </Provider>
        ,
      </React.StrictMode>
    );
    expect(tree).toMatchSnapshot();
  });
});
