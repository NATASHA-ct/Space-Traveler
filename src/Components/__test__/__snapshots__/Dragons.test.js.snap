// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is item 1`] = `
<div
  className="card dragonsContainer"
>
  <div
    className="dragonImg"
  >
    <img
      alt="dragon-img"
    />
  </div>
  <div
    className="dragonDescription"
  >
    <h1
      className="dragonName"
    />
    <div
      className="dragonType"
    />
    <button
      className=" "
      onClick={[Function]}
      type="button"
    >
      Reserve Dragon
    </button>
  </div>
</div>
`;
